www_src/pages/project-settings/project-settings.jsx

```diff
@@ -1,21 +1,83 @@
-var React = require('react');
+var React = require('react/addons');
+
+var api = require('../../lib/api');
 var render = require('../../lib/render.jsx');
+var router = require('../../lib/router.jsx');
 var TextInput = require('../../components/text-input/text-input.jsx');
 
 var ProjectSettings = React.createClass({
-  onDoneClick: function () {
-    // TODO : Wire up to Android menu
-    console.log(this.refs.title.validate());
-    console.log(this.refs.title.state.text);
+  mixins: [
+    React.addons.LinkedStateMixin,
+    router
+  ],
+  getInitialState: function () {
+    return {
+      title: ''
+    };
+  },
+  componentWillMount: function () {
+    var _this = this;
+
+    // Build up URI for API requests
+    var params = this.state.params;
+    var uri = '/users/1/projects/' + params.project;
+
+    // Fetch a fresh copy of the project's metadata & update state
+    api({
+      uri: uri
+    }, function (err, body) {
+      if (err) {
+        // @todo Handle error state (GH-1922)
+      }
+
+      _this.setState({
+        uri: uri,
+        title: body.project.title
+      });
+
+      _this.refs.title.setState({
+        inputLength: body.project.title.length
+      });
+    });
   },
   render: function () {
     return (
       <div id="projectSettings">
-        {/* TODO : Replace with header button in Android */}
-        <button onClick={this.onDoneClick} style={{marginBottom: "20px"}}>✓</button>
-        <TextInput ref="title" label="Title" maxlength={25} minlength={4} />
+        <TextInput id="title" ref="title" label="Title" maxlength={25} minlength={4} linkState={this.linkState} />
+        <button className="btn" onClick={this.onDoneClick} style={{marginBottom: "20px"}}>Save</button>
       </div>
     );
+  },
+
+  /**
+   * Click handler for persisting changes to project settings.
+   *
+   * @return {void}
+   */
+  onDoneClick: function () {
```

Comment:
You might consider adding logic to see if a PATCH is already being sent to avoid a bunch of stacked requests if someone taps a bunch of times (or even double taps for some reason...).